--- ../work/imagebuilder-24.10.2-ath79-nand/target/linux/ath79/image/nand.mk	2025-06-24 04:40:36.000000000 +0800
+++ target-linux-ath79-image-nand.mk	2025-08-30 15:49:41.036729269 +0800
@@ -366,6 +366,9 @@
 endef
 TARGET_DEVICES += netgear_r6100
 
+ifeq (,$(CUSTOM_IMAGE_SIZE))
+
+# default official openwrt wndr3700-v4 setting
 define Device/netgear_wndr3700-v4
   SOC := ar9344
   DEVICE_MODEL := WNDR3700
@@ -375,6 +378,25 @@
   NETGEAR_HW_ID := 29763948+128+128
   $(Device/netgear_ath79_nand)
 endef
+
+else
+
+# use cuttom wndr3700-v4 setting
+# 	$$() -> $() when using
+# 	DTS_CPPFLAGS not checked/used
+define Device/netgear_wndr3700-v4
+  SOC := ar9344
+  DEVICE_MODEL := WNDR3700
+  DEVICE_VARIANT := v4
+  UIMAGE_MAGIC := 0x33373033
+  NETGEAR_BOARD_ID := WNDR3700v4
+  NETGEAR_HW_ID := 29763948+128+128
+  $(Device/netgear_ath79_nand)
+  IMAGE_SIZE := $$(CUSTOM_IMAGE_SIZE)
+  DTS_CPPFLAGS := -D__CUSTOM_IMAGE_SIZE__
+endef
+
+endif
 TARGET_DEVICES += netgear_wndr3700-v4
 
 define Device/netgear_wndr4300
